app/src/main/java/com/duckduckgo/app/browser/downloader/FilenameExtractor.kt

```diff
@@ -74,13 +75,13 @@ class FilenameExtractor @Inject constructor(
         return guessedFilename
     }
 
-    private fun bestGuess(guesses: Guesses): String {
+    private fun bestGuess(guesses: Guesses): FilenameExtractionResult {
         val guess = guesses.bestGuess ?: guesses.latestGuess
         if (!guess.contains(".")) {
             pixel.fire(AppPixelName.DOWNLOAD_FILE_DEFAULT_GUESSED_NAME)
-            return guess + DEFAULT_FILE_TYPE
+            return FilenameExtractionResult.Failure(guess)
```

Comment:
I think this is a bit misleading. Normally, when modelling `Success` or `Failure`, the `Failure` contains the reason of failure. Here it contains a default value.
Why did we need to change the return type from `String` with the best guess, to a `Success` / `Failure` that anyway contains the best guess?